### PR TITLE
Allow multiple calls for hardware keyrings apis

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/19490

- Return CommandInProgress/BridgeNotReady error codes from transport layer for Trezor
- Return TransportLocked for Ledger
- Do not close Panel for CommandInProgress and TransportLocked errors
- Little codestyle improvements

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Need to check import, signing transactions/messages for both Ledger and Trezor
- Multiple clicks by buttons Sign/Confirm in the Panel should not affect the result

### CODE REVIEW

File: components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts [hunk 1]
```diff
@@ -7,44 +7,73 @@
...
+        data === TrezorErrorsCodes.CommandInProgress) {
+      return this.createErrorFromCode(data)
     }
-    if (!this.isUnlocked() && !(await this.unlock())) {
-      return new Error(getLocale('braveWalletUnlockError'))
+    const response: UnlockResponse = data.payload
+    this.unlocked = response.success
+    if (!response.success) {
+      const error = response.payload && response.payload.error ? response.payload.error : getLocale('braveWalletUnlockError')
+      const code = response.payload && response.payload.code ? response.payload.code : ''
```
Comment (onyb) [inline components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts:62]:
```suggestion
      const error = response.payload ?? getLocale('braveWalletUnlockError')
      const code = response.payload?.code ?? ''
```

**Review by onyb — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/common/hardware_operations.ts [hunk 2]
```diff
@@ -12,7 +13,7 @@ export interface SignatureVRS {
   s: string
 }
 export type HardwareOperationResult = {
-  success: boolean
+  success: Boolean
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_wallet_ui/common/hardware_operations.ts:16]: it alerts in another places where  `boolean` is used, changed it to be same in all places

File: components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts (hunk 1 above)
Comment (spylogsster) [reply components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts]: done

File: components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts [hunk 3]
```diff
@@ -7,44 +7,73 @@
...
 import { hardwareDeviceIdFromAddress } from '../hardwareDeviceIdFromAddress'
-import { SignHardwareMessageOperationResult, SignHardwareTransactionOperationResult } from '../../common/hardware_operations'
+import { GetAccountsHardwareOperationResult, HardwareOperationResult, SignHardwareMessageOperationResult, SignHardwareTransactionOperationResult } from '../../common/hardware_operations'
 
 export default class TrezorBridgeKeyring extends EventEmitter {
+  private unlocked: Boolean = false
   constructor () {
     super()
-    this.unlocked_ = false
   }
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts]: done

File: components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts [hunk 4]
```diff
@@ -7,44 +7,73 @@
...
+  UnlockResponse,
+  SignTransactionResponse,
+  SignMessageResponse,
+  TrezorGetAccountsResponse
 } from '../../common/trezor/trezor-messages'
 import { sendTrezorCommand } from '../../common/trezor/trezor-bridge-transport'
 import { getLocale } from '../../../common/locale'
 import { hardwareDeviceIdFromAddress } from '../hardwareDeviceIdFromAddress'
-import { SignHardwareMessageOperationResult, SignHardwareTransactionOperationResult } from '../../common/hardware_operations'
+import { GetAccountsHardwareOperationResult, HardwareOperationResult, SignHardwareMessageOperationResult, SignHardwareTransactionOperationResult } from '../../common/hardware_operations'
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts]: done

File: components/brave_wallet_ui/common/trezor/trezor-bridge-transport.ts [hunk 5]
```diff
@@ -68,13 +71,13 @@ export class TrezorBridgeTransport extends MessagingTransport {
     })
   }
 
-  private readonly getBridge = (): HTMLIFrameElement | null => {
-    return document.getElementById(this.frameId) as HTMLIFrameElement | null
+  private readonly hasBridgeCreated = (): Boolean => {
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_wallet_ui/common/trezor/trezor-bridge-transport.ts]: done

File: components/brave_wallet_ui/common/trezor/trezor-bridge-transport.ts [hunk 6]
```diff
@@ -68,13 +71,13 @@ export class TrezorBridgeTransport extends MessagingTransport {
...
+    return document.getElementById(this.frameId) != null
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_wallet_ui/common/trezor/trezor-bridge-transport.ts]: done

File: components/brave_wallet_ui/common/hardware_operations.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_wallet_ui/common/hardware_operations.ts]: done

File: components/brave_wallet_ui/common/async/hardware.ts [hunk 7]
```diff
@@ -29,6 +30,10 @@ export async function signTrezorTransaction (apiProxy: WalletApiProxy, path: str
   const signed = await deviceKeyring.signTransaction(path, txInfo, chainId.chainId)
   if (!signed || !signed.success || !signed.payload) {
     const error = signed.error ? signed.error : getLocale('braveWalletSignOnDeviceError')
+    const code = signed.code ? signed.code : undefined
+    if (code === TrezorErrorsCodes.CommandInProgress) {
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_wallet_ui/common/async/hardware.ts]: done

**Review by onyb — APPROVED** [review]
LGTM!

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.34.41 Chromium: 96.0.4664.45 (Official Build) nightly (arm64)
-- | --
Revision | `76e4c1bb2ab4671b8beba3444e61c0f17584b2fc-refs/branch-heads/4664@{#947}`
OS | macOS Version 12.0.1 (Build 21A559)

- Verified able to successfully connect hardware wallets (both Ledger and Trezor)
- Verified hardware wallet accounts are added to the Secondary accounts list
- Verified multiple clicks on confirm/sign button doesn't close the wallet panel and allows to sign messages or complete the transaction

**Pull request merged**